File: backend/OnboardingDiary.Application/Tasks/Dtos/TaskDto.cs
namespace OnboardingDiary.Application.Tasks.Dtos;

public record TaskDto(
    Guid Id,
    Guid UserId,
    DateTime Date,
    string Title,
    string? Description,
    string Category,
    string Status,
    string Priority,
    DateTime? CompletedAt,
    DateTime CreatedAt,
    DateTime UpdatedAt);

File: src/main/java/com/codev/onboardingdiary/web/dto/TaskDto.java
package com.codev.onboardingdiary.web.dto;

import com.codev.onboardingdiary.domain.Priority;
import com.codev.onboardingdiary.domain.TaskStatus;
import java.time.LocalDate;

public record TaskDto(Long id,
                      LocalDate date,
                      String title,
                      String description,
                      String category,
                      TaskStatus status,
                      Priority priority) {
}
